barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -3596,6 +3631,16 @@ ReturnDataError AvmTraceBuilder::op_return(uint8_t indirect, uint32_t ret_offset
         error = AvmError::CHECK_TAG_ERROR;
     }
 
+    bool is_top_level = true;
+    // Is it ok to write success here?
+    if (current_ext_call_ctx.context_id != 0) {
+        is_top_level = false;
+        current_ext_call_ctx = external_call_ctx_stack.top();
+        external_call_ctx_stack.pop();
+        set_call_ptr(uint8_t(current_ext_call_ctx.context_id));
+        write_to_memory(current_ext_call_ctx.success_offset, FF::one(), AvmMemoryTag::U1);
+    }
```

Comment:
since we have popped already - technically we are already in the parent context when we write. But i agree that "writing" in the child's opcode execution is cheating.

we'd probably need to abstract the context details from the opcode execution